app/controllers/SurveyController.java

```diff
@@ -171,7 +171,7 @@ public Result deleteSurvey(String surveyGuid, String createdOnString) throws Exc
         Survey survey = surveyService.getSurvey(keys);
         verifySurveyIsInStudy(session, studyId, survey);
         
-        surveyService.deleteSurvey(studyId, survey);
+        surveyService.deleteSurvey(session.getUser().getHealthCode(), studyId, survey);
```

Comment:
This is the researcher's healthcode. This is later used in surveyDao to check whether the survey already has responses, but since it's keyed off the healthcode of the participant, this check will always wrongfully say the survey has no responses.